spec/models/info_request_spec.rb

```diff
@@ -50,13 +50,254 @@
+        info_request = FactoryGirl.create(:info_request, attrs)
+        email, raw_email = email_and_raw_email(:raw_email => spam_email)
+        info_request.receive(email, raw_email, true)
+        expect(info_request.incoming_messages.size).to eq(1)
+      end
+    end
+
+    context 'handling rejected responses' do
+
+      it 'bounces rejected responses if the mail has a from address' do
+        attrs = { :allow_new_responses_from => 'nobody',
+                  :handle_rejected_responses => 'bounce' }
+        info_request = FactoryGirl.create(:info_request, attrs)
+        email, raw_email = email_and_raw_email(:from => 'bounce@example.com')
+        info_request.receive(email, raw_email)
+        bounce = ActionMailer::Base.deliveries.first
+        expect(bounce.to).to include('bounce@example.com')
+        ActionMailer::Base.deliveries.clear
+      end
+
+      it 'does not bounce external requests' do
+        info_request = FactoryGirl.create(:external_request)
+        email, raw_email = email_and_raw_email(:from => 'bounce@example.com')
+        info_request.receive(email, raw_email)
+        expect(ActionMailer::Base.deliveries).to be_empty
+        ActionMailer::Base.deliveries.clear
+      end
+
+      it 'discards rejected responses if the mail has no from address' do
+        attrs = { :allow_new_responses_from => 'nobody',
+                  :handle_rejected_responses => 'bounce' }
+        info_request = FactoryGirl.create(:info_request, attrs)
+        email, raw_email = email_and_raw_email(:from => '')
+        info_request.receive(email, raw_email)
+        expect(ActionMailer::Base.deliveries).to be_empty
+        ActionMailer::Base.deliveries.clear
+      end
+
+      it 'sends rejected responses to the holding pen' do
+        attrs = { :allow_new_responses_from => 'nobody',
+                  :handle_rejected_responses => 'holding_pen' }
+        info_request = FactoryGirl.create(:info_request, attrs)
+        email, raw_email = email_and_raw_email
+        info_request.receive(email, raw_email)
+        expect(InfoRequest.holding_pen_request.incoming_messages.size).to eq(1)
+        expect(ActionMailer::Base.deliveries.size).to eq(1)
+        ActionMailer::Base.deliveries.clear
+      end
+
+      it 'discards rejected responses' do
+        attrs = { :allow_new_responses_from => 'nobody',
+                  :handle_rejected_responses => 'blackhole' }
+        info_request = FactoryGirl.create(:info_request, attrs)
+        email, raw_email = email_and_raw_email
+        info_request.receive(email, raw_email)
+        expect(ActionMailer::Base.deliveries).to be_empty
+        ActionMailer::Base.deliveries.clear
+      end
+
+      it 'raises an error if there is an unknown handle_rejected_responses' do
+        attrs = { :allow_new_responses_from => 'nobody' }
+        info_request = FactoryGirl.create(:info_request, attrs)
+        info_request.update_attribute(:handle_rejected_responses, 'unknown_value')
```

Comment:
Line is too long. [82/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.